app/src/main/java/org/thoughtcrime/securesms/database/MmsSmsDatabase.java

```diff
@@ -115,6 +116,81 @@ public MmsSmsDatabase(Context context, SQLCipherOpenHelper databaseHelper) {
     return null;
   }
 
+  public @Nullable MessageRecord getSentMessageFor(long timestamp, String serializedAuthor) {
+
+    try (Cursor cursor = queryTables(PROJECTION, MmsSmsColumns.NORMALIZED_DATE_SENT + " = " + timestamp, null, null)) {
+      MmsSmsDatabase.Reader reader = readerFor(cursor);
+
+      MessageRecord messageRecord;
+      boolean isOwnNumber = Util.isOwnNumber(context, serializedAuthor);
+
+      while ((messageRecord = reader.getNext()) != null) {
+        if (isOwnNumber && messageRecord.isOutgoing())
+        {
+          return messageRecord;
+        }
+      }
+    }
+    Log.w(TAG, "Could not find any message sent from us at timestamp: " + timestamp + " - returning null.");
```

Comment:
I wouldn't say it's common - but I hate stuff failing silently and then it's a PITA to track down because it _looks_ like nothing untoward happened. Have adjusted the log level to INFO and removed the specific timestamp value in case there are any data sensitivity issues.